src/SchmackoSwap.sol

```diff
@@ -64,14 +71,18 @@ contract SchmackoSwap is ERC165, ReentrancyGuard {
     /// @param creator The address of the seller
     /// @param askPrice The amount the seller is asking for in exchange for the token
     struct Listing {
-        ERC1155Supply tokenContract;
+        IERC1155Supply tokenContract;
```

Comment:
why do we use ERC1155 here?

Reply:
Schmackoswap is a generic IERC1155 sales contract.